pageserver/src/repository.rs

```diff
@@ -167,15 +217,159 @@ pub trait Timeline {
     fn advance_last_record_lsn(&self, lsn: Lsn);
     fn get_last_record_lsn(&self) -> Lsn;
 
+    //
+    // Wait until WAL has been received up to the given LSN.
+    //
+    fn wait_lsn(&self, lsn: Lsn) -> Result<Lsn>;
+
     /// Get range [begin,end) of stored blocks. Used mostly for SMGR pseudorelations
     /// but can be also applied to normal relations.
-    fn get_range(&self, rel: RelTag, lsn: Lsn) -> Result<(u32, u32)>;
+    fn get_range(&self, rel: RelTag, lsn: Lsn) -> Result<(u32, u32)> {
```

Comment:
I wonder if we could use half-open [`Range<u32>`](https://doc.rust-lang.org/std/ops/struct.Range.html) from std instead of `(u32, u32)`